<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Footnotes

In [ ]:
while True:
    print('*', end='') #Click on the "Interrupt" button at the top of the page.

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

In [4]:
print = 'PRINT'

In [2]:
print('Hello World') #Click on the "Restart" button which is also next to the interrupt button

Hello World


Both Restart and Interrupt buttons can be found at the top of the page.

The Restart Button clears all variables 

The interrupt Button stops the current code that is running 

In [7]:
%%timeit

squares = [i**2 for i in range(10)]

660 ns ± 79 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [8]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

780 ns ± 52.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The code was ran 1,000,000 times (referred to as 1,000,000 loops), 7 times.

780ns refers to the mean time taken within the 7 times to run the 1 loop.

$\pm$ 52.7 ns refers to the standard devitation within the 7 times to run the 1 loop.

where ns is nanoseconds

In [3]:
Squares = ([i**2 for i in range(10)])
print (Squares)

squares = list()
for i in range(10):
    squares.append(i**2)
print (squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
